docs/plain-bundle-spec.md

```diff
@@ -0,0 +1,138 @@
+# Plain Bundle Spec
+
+## Overview
+
+This document is meant to define the plain bundle format as a reference for those publishing plain bundles for use with
+RukPak. A bundle is a collection of Kubernetes resources that are packaged together for the purposes of installing onto
+a Kubernetes cluster. A bundle can be unpacked onto a running cluster, where controllers can then create the underlying
+content embedded in the bundle. The bundle can be used as the underlying `spec.source` for a [Bundle](/README.md#Bundle)
+resource.
+
+A plain bundle is simply a collection of static, arbitrary, Kubernetes YAML manifests in a given directory. A plain
+bundle can consist of a container image, a directory in a git repository, or any other content source that
+the [plain bundle provisioner](/internal/provisioner/plain/README.md) supports.
+
+The currently implemented plain bundle format is the `plain+v0` format. The name of the bundle format, `plain+v0`
```

Comment:
I've been thinking about our naming. I wonder if we should reuse the `metav1.Object` concept of using apiVersion and kind (e.g. `apiVersion: core.rukpak.io/v0, kind: Plain`).

It gives us everything we need:
- explicit namespacing with a domain name-based group name
- major versioning
- different kinds of bundles
- extensibility (anyone can create a bundle format and a provisioner in their own group)

It also gives us a more structured type to store all of this metadata, so that we don't have to argue over "is it plain.v0, plain+v0, or plain-v0?"

Reply:
I do like this idea, I think it's intuitive. Using `Kind` to specify different types of Bundles is nice UX too.